mantidproject/mantid
# Reflectometry GUI restructuring 2: Save tab

### PR ~~#23630~~ should be merged first
This can be reviewed in the meantime by comparing against [22263_refl_gui_instancing](https://github.com/gemmaguest/mantid/tree/22263_refl_gui_instancing)

**Description of work.**

Ongoing work to integrate the `JobTreeView` into the `ISIS Reflectometry` GUI. This PR is being merged to a feature branch because not all functionality is implemented yet.

This PR restructures the Save Tab view/presenter for the GUI to follow the same structure as the other tabs which have already been restructured as part of #23630.

- Rename the `...ReflSaveTab...` components to just `Save...`
- Move the components to the `GUI/Save/` subdirectory
- Add a new `SaveViewSubscriber` interface and use this inside the view rather than the presenter.
- Tidy up some unrequired `#include`s and add some doxygen comments
- Tidy up the existing mock objects and remove things that are no longer used

Note that MVP models have not been added at this point but this will be done in a later PR.

**Report to:** nobody

**To test:**

- Open the `ISIS Reflectometry` interface
- Go to the `Save` tab
- Load a run or two e.g. `INTER13460`
- Click `Refresh` - the run(s) should appear in the workspace list
- Click `Save` and check that the workspace is saved in the specified save path
- Try some of the other options and check they behave sensibly.
- Note that `Save ASCII on completion` currently does nothing because processing is not implemented yet.

Refs #23051

*This does not require release notes* because it involves under-the-hood changes only.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
